proxy/http/HttpSM.cc

```diff
@@ -3965,13 +3965,59 @@ HttpSM::state_remap_request(int event, void * /* data ATS_UNUSED */)
   return 0;
 }
 
+// This check must be called before remap.  Otherwise, the client_request host
+// name may be changed.
+void
+HttpSM::check_sni_host()
+{
+  // Check that the SNI and host name fields match, if it matters
+  // Issue warning or mark the transaction to be terminated as necessary
+  int host_len;
+  const char *host_name = t_state.hdr_info.client_request.host_get(&host_len);
+  if (host_name && host_len) {
+    if (ua_txn->support_sni()) {
+      int host_sni_policy   = t_state.http_config_param->http_host_sni_policy;
+      NetVConnection *netvc = ua_txn->get_netvc();
+      if (netvc) {
+        IpEndpoint ip = netvc->get_remote_endpoint();
+        if (SSLNetVConnection::TestClientAction(std::string{host_name, static_cast<size_t>(host_len)}.c_str(), ip,
```

Comment:
It doesn't seem like SSLNetVConnection is an appropriate class to have this function (TestClientAction)? SNI can be used with QUIC connections and it's just not implemented yet.

Reply:
Could move TestClientAction to SNIConfig.